### PR TITLE
Feature/nl favorite functionality

### LINKED ISSUE #111: Favoriting functionality.

### PR DESCRIPTION
**Pull request by NicoleLatifi**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features

### Detailed Description / What does this PR do?

Created functionality that allows a logged-in user to favorite and unfavorite movies from both the home page and detailed movie page by clicking on a heart icon. These favorited movies can be viewed on the Favorite Movies page.

### Files modified:

src/App/App.js 
src/Favorite-Movies/Favorite-Movies.css 
src/Favorite-Movies/Favorite-Movies.js 
src/Fetch.js 
src/Header/Header.css 
src/Header/Header.js 
src/Login/Login.css 
src/Movie-Card/Movie-Card.css 
src/Movie-Card/Movie-Card.js 
src/Movie-Page/Movie-Page.css 
src/Movie-Page/Movie-Page.js 
src/Movie-Section/Movie-Section.css 
src/Movie-Section/Movie-Section.js 

### Has This Been Tested?
- [ ] No
- [x] Yes

### Are there specific methods for testing?

Open in browser and, with the console open, login and click through favoriting and unfavoriting movies and viewing them on the Favorite Movies page.

### What are the relevant tickets?

resolves #111 

### Message/Questions for reviewer:

I will write tests for this next.

### Checklist:
- [x] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Movie-Card/Movie-Card.js [hunk 1]
```diff
@@ -1,35 +1,80 @@
...
+            <img className='heart-icon' src={unfilledHeart} alt='Unselected heart' onClick={this.favoriteMovie} />
+          </div>
+        }
+        {currentUser && favorited &&
+          <div className='heart-background'>
+            <img className='heart-icon' src={filledHeart} alt='Selected heart' onClick={this.favoriteMovie} />
+          </div>
+        }
+        <Link to={`/movies/${movie.id}`}>
+          <div className='clickable'>
```
Comment (NicoleLatifi) [inline src/Movie-Card/Movie-Card.js:60]: This div purely exists to be the clickable area of the movie card which allows the user to click the heart icon without clicking the movie card. It's a very rough work-around.

File: src/Movie-Section/Movie-Section.js [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 import PropTypes from 'prop-types';
 
-const MovieSection = ({ movies, currentUser }) => {
+const MovieSection = ({ movies, currentUser, favoriteMovies, addFavoriteMovie, getUsersFavoriteMovies }) => {
   const movieCards = movies.map((movie) => {
     return (
-      <Link to={`/movies/${movie.id}`} key={movie.id}>
-        <MovieCard movie={movie} currentUser={currentUser} />
-      </Link>
+        <MovieCard
```
Comment (NicoleLatifi) [inline src/Movie-Section/Movie-Section.js:9]: I moved the `<Link>` to the MoveCard.

File: src/App/App.js [hunk 3]
```diff
@@ -52,6 +54,13 @@ class App extends Component {
     this.setState({ currentUser: false });
   };
 
+  getUsersFavoriteMovies = () => {
+    getFavoriteMovies()
+      .then((favoriteMovies) => {
+        this.setState({ favoriteMovies });
+      })
```
Comment (relyt4me) [inline src/App/App.js:61]: We might want a catch here eventually

File: src/Favorite-Movies/Favorite-Movies.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      {!favoriteMovies.length && <p className='no-favorites-message'>You don't have any favorited movies yet. Go back to start favoriting!</p>}
+      <div className='favorite-movies-container'>{movieCards}</div>
+    </section>
+  );
+};
+
+export default FavoriteMovies;
+
+FavoriteMovies.propTypes = {
+  movies: PropTypes.array,
```
Comment (relyt4me) [inline src/Favorite-Movies/Favorite-Movies.js:35]: Good continuation of proptypes

File: src/Movie-Card/Movie-Card.css [hunk 5]
```diff
@@ -1,10 +1,35 @@
...
+  width: 34px;
+}
+
+.heart-icon {
+  height: 24px;
+  width: 24px;
+  margin-left: 3px;
+  margin-top: 3px;
+}
+
```
Comment (relyt4me) [inline src/Movie-Card/Movie-Card.css:32]: This is great looking

**Pull request merged**